airbyte-workers/src/main/java/io/airbyte/workers/helper/FailureHelper.java

```diff
@@ -34,18 +35,62 @@ public static FailureReason genericFailure(final Throwable t, final Long jobId,
         .withMetadata(jobAndAttemptMetadata(jobId, attemptNumber));
   }
 
+  public static FailureReason genericFailure(final AirbyteTraceMessage m, final Long jobId, final Integer attemptNumber) {
+    return new FailureReason()
+        .withInternalMessage(m.getError().getInternalMessage())
+        .withStacktrace(m.getError().getStackTrace())
+        .withTimestamp(m.getEmittedAt().longValue())
+        .withMetadata(traceMessageMetadata(jobId, attemptNumber));
```

Comment:
We also need to set the `failureType` on the FailureReason, based on the `AirbyteErrorTraceMessage`'s `failureType` value (accounting for if that value is `null`, then this should default to `system_error`)

Reply:
I actually wasn't sure where we landed on adding a default value here - do we want to give the failure reason a default of system_error or leave it null?

Reply:
working on this now (still have the question about the default vs leaving it null), but it's not actually straightforward to convert one enum into another one. I'm converting the trace message enum into a string, and then using .valueOf(string) to convert it into the FailureReason FailureType enum value, but I'm not sure this is a great way to do this. this throws an exception of .valueOf is passed anything that isn't listed FailureType enum, so if a new enum is ever added to AirbyteErrorTraceMessage and a corresponding enum is not added to FailureType on FailureReason, then we'll have to catch exceptions here, and I'm not sure that's a great pattern

Reply:
@alovew yeah in this doc there was a note that it should default to system_error if missing: https://docs.google.com/document/d/1ctrj3Yh_GjtQ93aND-WH3ocqGxsmxyC3jfiarrF6NY0/edit#heading=h.uupw5avb79b0